File: 39-failover/npm-package/failover.ts
import axios from 'axios';
const API = 'https://cost-katana-backend.store/api'\;
const KEY = process.env.COST_KATANA_API_KEY;

export async function getFailoverConfig() {
  const res = await axios.get(`${API}/failover/config`,
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log('🔄 Failover config:', res.data.data);
  return res.data.data;
}

export async function updateFailoverConfig(enabled: boolean, providers: string[]) {
  const res = await axios.put(`${API}/failover/config`, {enabled, providers},
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log('✅ Failover config updated');
  return res.data.data;
}

File: 39-failover/python-sdk/failover.py
import os, requests
API = "https://api.costkatana.com/api"
KEY = os.getenv("COST_KATANA_API_KEY")

def get_failover_config():
    res = requests.get(f"{API}/failover/config",
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print(f"🔄 Failover config: {res.json()['data']}")
    return res.json()["data"]

def update_failover_config(enabled, providers):
    res = requests.put(f"{API}/failover/config",
        json={"enabled": enabled, "providers": providers},
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print("✅ Failover config updated")
    return res.json()["data"]
